File: web/ui/stores/jobs.ts
import { create } from "zustand";
import { api } from "@/lib/api";
import type { JobSummary, JobDetail, JobListResponse, SortDir } from "@/lib/types";

interface JobFilters {
  stage: string;
  minScore: number | null;
  maxScore: number | null;
  site: string;
  search: string;
  sortBy: string;
  sortDir: SortDir;
  page: number;
  limit: number;
}

interface JobStore {
  jobs: JobSummary[];
  total: number;
  pages: number;
  filters: JobFilters;
  selectedJob: JobDetail | null;
  loading: boolean;
  fetchJobs: () => Promise<void>;
  setFilters: (f: Partial<JobFilters>) => void;
  selectJob: (url: string) => Promise<void>;
  clearSelection: () => void;
}

export const useJobStore = create<JobStore>((set, get) => ({
  jobs: [],
  total: 0,
  pages: 0,
  filters: {
    stage: "",
    minScore: null,
    maxScore: null,
    site: "",
    search: "",
    sortBy: "fit_score",
    sortDir: "desc",
    page: 1,
    limit: 50,
  },
  selectedJob: null,
  loading: false,

  fetchJobs: async () => {
    set({ loading: true });
    try {
      const f = get().filters;
      const data = await api.getJobs({
        stage: f.stage || undefined,
        min_score: f.minScore ?? undefined,
        max_score: f.maxScore ?? undefined,
        site: f.site || undefined,
        search: f.search || undefined,
        sort_by: f.sortBy,
        sort_dir: f.sortDir,
        page: f.page,
        limit: f.limit,
      });
      set({ jobs: data.jobs, total: data.total, pages: data.pages });
    } finally {
      set({ loading: false });
    }
  },

  setFilters: (f) => {
    set((s) => ({ filters: { ...s.filters, ...f, page: f.page ?? 1 } }));
  },

  selectJob: async (url) => {
    try {
      const job = await api.getJob(url);
      set({ selectedJob: job });
    } catch { /* ignore */ }
  },

  clearSelection: () => set({ selectedJob: null }),
}));

File: web/api/routers/jobs.py
"""Jobs API router."""

from __future__ import annotations

from fastapi import APIRouter, HTTPException, Query
from typing import Optional

from web.api.models.job import (
    JobActionResponse,
    JobCoverLetterResponse,
    JobDeleteResponse,
    JobDetail,
    JobListResponse,
    JobResumeResponse,
    ScoreOverride,
)
from web.api.services import job_service

router = APIRouter(prefix="/api/jobs", tags=["jobs"])


@router.get("", response_model=JobListResponse)
async def list_jobs(
    stage: Optional[str] = Query(None),
    min_score: Optional[int] = Query(None),
    max_score: Optional[int] = Query(None),
    site: Optional[str] = Query(None),
    search: Optional[str] = Query(None),
    sort_by: str = Query("fit_score"),
    sort_dir: str = Query("desc"),
    page: int = Query(1),
    limit: int = Query(50),
) -> JobListResponse:
    """List jobs with filtering, sorting, and pagination."""
    return job_service.get_job_list(
        stage=stage,
        min_score=min_score,
        max_score=max_score,
        site=site,
        search=search,
        sort_by=sort_by,
        sort_dir=sort_dir,
        page=page,
        limit=limit,
    )


@router.get("/{url:path}", response_model=JobDetail)
async def get_job(url: str) -> JobDetail:
    """Get full job details by URL."""
    job = job_service.get_job_detail(url)
    if not job:
        raise HTTPException(status_code=404, detail="Job not found")
    return job


@router.get("/{url:path}/resume", response_model=JobResumeResponse)
async def get_job_resume(url: str) -> JobResumeResponse:
    """Get original and tailored resume for a job."""
    result = job_service.get_job_resume(url)
    if not result:
        raise HTTPException(status_code=404, detail="Job not found")
    return result


@router.get("/{url:path}/cover-letter", response_model=JobCoverLetterResponse)
async def get_job_cover_letter(url: str) -> JobCoverLetterResponse:
    """Get cover letter for a job."""
    result = job_service.get_job_cover_letter(url)
    if not result:
        raise HTTPException(status_code=404, detail="Cover letter not found")
    return result


@router.put("/{url:path}/score", response_model=JobActionResponse)
async def update_score(url: str, body: ScoreOverride) -> JobActionResponse:
    """Manually override a job's score."""
    job = job_service.get_job_detail(url)
    if not job:
        raise HTTPException(status_code=404, detail="Job not found")
    job_service.update_job_score(url, body.score, body.reasoning)
    return JobActionResponse()


@router.delete("/{url:path}", response_model=JobDeleteResponse)
async def delete_job(url: str) -> JobDeleteResponse:
    """Delete a job."""
    job = job_service.get_job_detail(url)
    if not job:
        raise HTTPException(status_code=404, detail="Job not found")
    job_service.delete_job(url)
    return JobDeleteResponse()
